core/src/test/java/org/kohsuke/stapler/framework/io/LargeTextTest.java

```diff
@@ -53,13 +63,53 @@ public void writeLogTo() throws Exception {
             // right
         }
     }
+
     String tail(String text, long start) throws IOException {
-        ByteBuffer bb = new ByteBuffer();
-        bb.write(text.getBytes(), 0, text.length());
-        LargeText t = new LargeText(bb, true);
+        LargeText t;
+        try (ByteBuffer bb = new ByteBuffer()) {
+            bb.write(text.getBytes(), 0, text.length());
+
+            t = new LargeText(bb, true);
+        }
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         assertEquals(text.length(), t.writeLogTo(start, baos));
         return baos.toString();
     }
 
+    @Issue("#141")
+    @Test
+    public void writeLogToWithLargeFile() throws Exception {
+        Path path = Files.createTempFile("stapler-test", ".log.gz");
+        long size = Integer.MAX_VALUE + 256L;
+        writeLargeFile(path, size);
+
+        LargeText t = new LargeText(path.toFile(), StandardCharsets.US_ASCII, true);
```

Comment:
For [JEP-210](https://jenkins.io/jep/210) and friends I have been thinking that we anyway need to make `Source` and `Session` (or some slight variants) `public` and add a `LargeText` constructor taking `Source`. That would allow this test to be written in a way that does not actually consume any disk or RAM.

Reply:
Yes that would be good.  LargeText should be mockable, but the reliance on File gets in the way.